be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp

```diff
@@ -809,8 +825,26 @@ Status VerticalSegmentWriter::write_batch() {
         _olap_data_convertor->clear_source_content();
         _num_rows_written += data.num_rows;
     }
-
     _batched_blocks.clear();
+    _full_blocks.clear();
+    return Status::OK();
+}
+
+Status VerticalSegmentWriter::_make_full_block(std::shared_ptr<vectorized::Block>& block_ptr,
+                                               const vectorized::Block* block, size_t num_rows) {
+    block_ptr = std::make_shared<vectorized::Block>(_tablet_schema->create_block());
+    const auto& including_cids = _opts.rowset_ctx->partial_update_info->update_cids;
+    size_t input_id = 0;
+    for (auto i : including_cids) {
+        block_ptr->replace_by_position(i, block->get_by_position(input_id++).column);
+    }
+    std::vector<bool> use_default_or_null_flag;
```

Comment:
different with mow partial update, we should use null if possible, if the column is not nullable, then use defalt value